core/src/options/mod.rs

```diff
@@ -43,7 +44,7 @@ impl FromMeta for DefaultExpression {
     }
 
     fn from_string(lit: &str) -> Result<Self> {
-        Ok(DefaultExpression::Explicit(syn::Path::from(lit)))
+        Ok(DefaultExpression::Explicit(syn::Path::from(syn::Ident::new(lit, Span::call_site()))))
```

Comment:
This changes the meaning of this line; we allow `path::to::function` here, but I don't think that's a valid ident.

Reply:
That's concerning. This is supposed to have behaved like `serde(default="path::to::foo")`; I wonder if it never worked. It's worth adding a test that exercises the path functionality.